applications/convection_diffusion_application/python_scripts/convection_diffusion_base_solver.py

```diff
@@ -0,0 +1,760 @@
+            if convention_diffusion_settings.IsDefinedVelocityVariable():
+                self.main_model_part.AddNodalSolutionStepVariable(convention_diffusion_settings.GetVelocityVariable())
+            if convention_diffusion_settings.IsDefinedSpecificHeatVariable():
+                self.main_model_part.AddNodalSolutionStepVariable(convention_diffusion_settings.GetSpecificHeatVariable())
+            if convention_diffusion_settings.IsDefinedReactionVariable():
+                self.main_model_part.AddNodalSolutionStepVariable(convention_diffusion_settings.GetReactionVariable())
+        else:
+            raise Exception("The provided main_model_part does not have CONVECTION_DIFFUSION_SETTINGS defined.")
+        
+        # Adding nodal area variable (some solvers use it. TODO: Ask)
+        #self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NODAL_AREA)
+        # If LaplacianElement is used
+        if (self.settings["element_replace_settings"]["element_name"].GetString() == "LaplacianElement"):
+            self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NORMAL)
+        
+        self.print_on_rank_zero("::[ConvectionDiffusionBaseSolver]:: ", "Variables ADDED")
+
+    def GetMinimumBufferSize(self):
+        return 2
+
+    def AddDofs(self):
+        # this can safely be called also for restarts, it is internally checked if the dofs exist already
+        settings = self.main_model_part.ProcessInfo[KratosMultiphysics.CONVECTION_DIFFUSION_SETTINGS]
+        if settings.IsDefinedReactionVariable():
+            KratosMultiphysics.VariableUtils().AddDof(settings.GetUnknownVariable(), settings.GetReactionVariable(),self.main_model_part)
+        else:
+            KratosMultiphysics.VariableUtils().AddDof(settings.GetUnknownVariable(), self.main_model_part)
+        self.print_on_rank_zero("::[ConvectionDiffusionBaseSolver]:: ", "DOF's ADDED")
+
+    def ImportModelPart(self):
+        """This function imports the ModelPart
+        """
+        if self.solver_imports_model_part:
+            self._ImportModelPart(self.main_model_part, self.settings["model_import_settings"])
+
+    def PrepareModelPart(self):
+        if not self.is_restarted():
+            # Check and prepare computing model part and import constitutive laws.
+            self._execute_after_reading()
+
+            throw_errors = False
+            KratosMultiphysics.TetrahedralMeshOrientationCheck(self.main_model_part, throw_errors).Execute()
+            
+            KratosMultiphysics.ReplaceElementsAndConditionsProcess(self.main_model_part,self._get_element_condition_replace_settings()).Execute()
+        
+            self._set_and_fill_buffer()
+        
+        # This will be removed once the Model is fully supported! => It wont e necessary anymore
+        if not self.model.HasModelPart(self.main_model_part.Name):
+            self.model.AddModelPart(self.main_model_part)
+
+        print(self.model)
```

Comment:
without echo_lvl?

Reply:
Maybe there is a leftover from the debug